Simplify framework install docs (#44900)
dotnet/docs: docs/framework/install/on-windows-and-server.md

---
title: Install .NET Framework on Windows
description: Learn how to install .NET Framework on Windows 11, Windows 10, and Windows Server. This article also includes information about .NET Framework and unsupported versions of Windows, such as Windows 8, Windows Vista, and Windows XP.

> ms.date: 02/12/2025

Reviewer: Should we add ms.custom:updateeachrelease since this article can get out of date easily?

Author: This will just become part of my normal OS updates. I'll have to watch the Server/Client releases for Windows. There shouldn't be anymore .NET Framework releases.

---
# Install .NET Framework on Windows and Windows Server

.NET Framework is included on all current versions of Windows and Windows Server. This article helps you understand which version of .NET Framework is included in Windows and Windows Server, and if an upgrade is available.

> The latest version of .NET Framework is 4.8.1. This version of .NET Framework supports all .NET Framework 4 apps, even if that app was created for an older version of .NET Framework 4. For more information about a specific release of .NET Framework, see [.NET Framework versions and dependencies](versions-and-dependencies.md).

Reviewer: What does "an older version of .NET Framework 4" mean?

Author: Would it be clearer with Older release? Prior release? Prior version? You can't and don't need to downgrade the version of .NET Framework on your machine to run an app that targets an older version.

Author: Sure. I'll trim this and simplify it

> [!NOTE]
> .NET Framework is a Windows-only technology, and is separate from .NET (formerly called .NET Core). For more information, see [Introduction to .NET](../../core/introduction.md).

## Supported versions of .NET Framework

The following versions of .NET Framework are still supported:

- [.NET Framework 4.8.1](https://dotnet.microsoft.com/download/dotnet-framework/net481)
- [.NET Framework 4.8](https://dotnet.microsoft.com/download/dotnet-framework/net48)
- [.NET Framework 4.7.2](https://dotnet.microsoft.com/download/dotnet-framework/net472)
- [.NET Framework 4.7.1](https://dotnet.microsoft.com/download/dotnet-framework/net471)
- [.NET Framework 4.7](https://dotnet.microsoft.com/download/dotnet-framework/net47)
- [.NET Framework 4.6.2](https://dotnet.microsoft.com/download/dotnet-framework/net462) (support ends January 12, 2027)
- [.NET Framework 3.5 Service Pack 1](https://dotnet.microsoft.com/download/dotnet-framework/net35-sp1) (support ends January 9, 2029)

### .NET Framework 3.5

> .NET Framework 3.5 is still supported by Microsoft, even though it's an older version of .NET Framework. However, only the .NET Framework 3.5 runtime is supported, which runs apps. Developing new apps that target .NET Framework 3.5 isn't supported. This version of .NET Framework supports versions 1.0 through 3.5, and can be installed alongside .NET Framework 4.

Reviewer: ```suggestion
.NET Framework 3.5 is still supported by Microsoft, even though it's an older version of .NET Framework. However, only the .NET Framework 3.5 runtime is supported, which runs apps. Developing new apps that target .NET Framework 3.5 isn't supported. This version of .NET Framework supports running apps that target versions 1.0 through 3.5, and can be installed alongside .NET Framework 4.
```

### Install .NET Framework on Windows 11

If you're using Windows 11 21H2, install .NET Framework 4.8.1 by downloading and running the installer. If you're using any other version of Windows 11, the latest .NET Framework is already installed.

> [!div class="button"]
> [Download .NET Framework 4.8.1](https://dotnet.microsoft.com/download/dotnet-framework/net481)

If you need to install .NET Framework 3.5, which supports .NET Framework apps 1.0 through 3.5, refer to the [.NET Framework 3.5 section](#net-framework-35).

For more downloads, see [All downloads](#all-downloads).

## Windows 10

.NET Framework 4.6 was originally included with Windows 10. However, newer releases of Windows 10 included upgraded versions of .NET Framework.

Windows 10 22H2 is the last supported version of Windows 10. Support ends October 14, 2025. For more information about Windows 10 end-of-support dates, see [Windows 10 Home and Pro Lifecycle](/lifecycle/products/windows-10-home-and-pro) and [Windows Lifecycle FAQ](/lifecycle/faq/windows).

> In the following table, ❌ represents an unsupported version of Windows 10 and ✔️ represents a supported version of Windows 10. The table also describes which version of .NET Framework is included with Windows 10, and which version of .NET Framework you can upgrade to.

Reviewer: ```suggestion
In the following table, ❌ represents an unsupported version of Windows 10 and ✔️ represents a supported version of Windows 10. The table also describes which version of .NET Framework is included with a particular Windows 10 version, and which version of .NET Framework you can upgrade to.
```

### Install .NET Framework on Windows 10

The latest version of .NET Framework is 4.8.1, which can be installed on Windows 10 22H2.

> [!div class="button"]
> [Download .NET Framework 4.8.1](https://dotnet.microsoft.com/download/dotnet-framework/net481)

If you need to install .NET Framework 3.5, which supports .NET Framework apps 1.0 through 3.5, refer to the [.NET Framework 3.5 section](#net-framework-35).

For more downloads, see [All downloads](#all-downloads).

## Windows Server

Windows Server, whether it's in support or not, comes with a version of .NET Framework. Only Windows Server 2022 and Windows Server 2025 are supported, and they both support the latest version of .NET Framework. For more information about Windows Server end-of-support dates, see the following articles:

- [Windows Server 2025 Lifecycle](/lifecycle/products/windows-server-2025)
- [Windows Server 2022 Lifecycle](/lifecycle/products/windows-server-2022)
- [Windows Server 2019 Lifecycle](/lifecycle/products/windows-server-2019)
- [Product Lifecycle Search Query - Windows Server](/lifecycle/products/?products=windows&terms=Windows%20Server)
- [Windows Lifecycle FAQ](/lifecycle/faq/windows).

> In the following table, ❌ represents an unsupported version of Windows 10 and ✔️ represents a supported version of Windows 10. The table also describes which version of .NET Framework is included with Windows 10, and which version of .NET Framework you can upgrade to.

Reviewer: ```suggestion
In the following table, ❌ represents an unsupported version of Windows Server and ✔️ represents a supported version of Windows Server. The table also describes which version of .NET Framework is included with a particular Windows Server version, and which version of .NET Framework you can upgrade to.
```

> 
> The following table describes which version of .NET Framework is included with Windows Server, and which version of .NET Framework you can upgrade to.

Reviewer: ```suggestion
```

- [.NET Framework 4.8](https://dotnet.microsoft.com/download/dotnet-framework/net48)
- [.NET Framework 4.7.2](https://dotnet.microsoft.com/download/dotnet-framework/net472)
- [.NET Framework 4.7.1](https://dotnet.microsoft.com/download/dotnet-framework/net471)
- [.NET Framework 4.7](https://dotnet.microsoft.com/download/dotnet-framework/net47)
- [.NET Framework 4.6.2](https://dotnet.microsoft.com/download/dotnet-framework/net462)
- [.NET Framework 4.6.1](https://dotnet.microsoft.com/download/dotnet-framework/net461)
- [.NET Framework 4.6](https://dotnet.microsoft.com/download/dotnet-framework/net46)
- [.NET Framework 4.5.2](https://dotnet.microsoft.com/download/dotnet-framework/net452)
- [.NET Framework 4.5.1](https://dotnet.microsoft.com/download/dotnet-framework/net451)
- [.NET Framework 4.5](https://dotnet.microsoft.com/download/dotnet-framework/net45)
- [.NET Framework 4.0](https://dotnet.microsoft.com/download/dotnet-framework/net40)
- [.NET Framework 3.5 Service Pack 1](https://dotnet.microsoft.com/download/dotnet-framework/net35-sp1)

Downloads of other versions of .NET Framework are no longer provided by Microsoft.

## See also

> - [Install the .NET Framework for developers](guide-for-developers.md)

Reviewer: ```suggestion
- [Install .NET Framework for developers](guide-for-developers.md)
```
